src/main/java/net/apnic/rdap/ip/IPStatsResourceLocator.java

```diff
@@ -46,7 +46,7 @@ public RDAPAuthority authorityForResource(IpRange range)
      * {@inheritDocs}
      */
     @Override
-    public void putResourceMapping(IpRange resource, RDAPAuthority authority)
+    public synchronized void putResourceMapping(IpRange resource, RDAPAuthority authority)
```

Comment:
You are going to always have a synchronisation point as the scrapers are multi threaded. Regardless of how that happens a scrapers result will need to be integrated.

But I do think there is a better method here than making the whole function a critical section. On top of that the reads are not synchronised.